.github/workflows/knative-dependabot.yaml

```diff
@@ -0,0 +1,36 @@
+name: depndabot - update-codegen.sh
+
+on:
+  pull_request:
+
+permissions:
+  contents: write
+
+jobs:
+  generate_some_code:
+    name: update deps
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+          path: ./src/github.com/${{ github.repository }}
+          fetch-depth: 0
```

Comment:
Curious on these settings some seem to be the defaults and why do you want all history?

Reply:
Just used with same value in `.github/workflows/knative-downstream.yaml`. It seems that it was added for some reason https://github.com/knative/pkg/pull/2265#discussion_r702107869
If we remove it, I think we should use consistent value in all action.